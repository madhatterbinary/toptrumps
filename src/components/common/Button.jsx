import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const {
    buttonClassName,
    buttonOnClick,
    buttonLabel,
    buttonIsDisabled,
  } = props;

  return (
    <button
      className={`btn ${buttonClassName} ${buttonIsDisabled && 'disabled'}`}
      onClick={buttonOnClick}
      disabled={buttonIsDisabled}
    >
      {buttonLabel}
    </button>
  );
};

Button.propTypes = {
  buttonClassName: PropTypes.string,
  buttonOnClick: PropTypes.func,
  buttonLabel: PropTypes.string,
  buttonIsDisabled: PropTypes.bool,
};

Button.defaultProps = {
  buttonClassName: '',
  buttonOnClick: () => {},
  buttonLabel: '',
  buttonIsDisabled: true,
};

export default Button;
