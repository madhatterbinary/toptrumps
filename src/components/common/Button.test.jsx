import React from 'react';
import Button from './Button';
import TestRenderer from 'react-test-renderer';

describe('Button', () => {
  it('renders', () => {
    const component = TestRenderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
