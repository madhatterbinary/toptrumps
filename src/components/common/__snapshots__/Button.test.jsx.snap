// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<button
  className="btn  disabled"
  disabled={true}
  onClick={[Function]}
>
  
</button>
`;
